### ISSUE TITLE
Memory leak during CPU inference

### ISSUE THREAD

**Issue by debasish-mihup** [burchim/EfficientConformer#7]
I have trained efficient conformer transducer and during inference on a CPU in a flask based web app, I see there is a memory leak at             

https://github.com/burchim/EfficientConformer/blob/2f59ed2564c45c5978742fe1984cd48363064c88/models/encoders.py#L128

x, attention, hidden = block(x, mask)

The memory used at the above line while running an inference is never released, thus causing OOM is some time. Increase in memory in each iteration gets lowered but not totally removed if I use **jemalloc**.

**Comment by debasish-mihup** [comment]
Using last release version of jemalloc 5.2.1 and using torch.no_grad() fixed the memory leak.

**Comment by burchim** [comment]
Hi,

Thanks for pointing this out!
The gradient of attention weights and hidden states aren't detached in attention layers.
